app/models/manageiq/providers/embedded_terraform/automation_manager/configuration_script_source.rb

```diff
@@ -85,22 +88,51 @@ def find_templates_in_git_repo
               .group_by          { |file| File.dirname(file) }
               .select            { |_dir, files| files.any? { |f| f.end_with?(".tf", ".tf.json") } }
               .transform_values! { |files| files.map { |f| File.basename(f) } }
-              .each do |parent_dir, files|
-        name = self.class.template_name_from_git_repo_url(git_repository.url, parent_dir)
+              .each do |relative_path, files|
+        name = self.class.template_name_from_git_repo_url(git_repository.url, relative_path)
+
+        template_full_path = File.join(git_checkout_tempdir, relative_path)
```

Comment:
I'm pretty sure git_checkout_tempdir is a Pathname (please verify) so you can simply `git_checkout_tempdir.join(relative_path)`

Reply:
```
undefined method `join' for "/var/folders/0j/t_rw5py52bl3yd8sx1bzd4280000gn/T/embedded-terraform-runner-git20240924-65302-v3ebjh":String

        template_full_path = git_checkout_tempdir.join(relative_path)
        
```
Can't use